gitpod-io/gitpod
# Limit connections

## Description
Introduce limiting of new connections created per minute. For more info see the [RFC](https://www.notion.so/gitpod/Rate-limit-network-connections-b3ac59e7db5d41198d63cc7f374c0333).

## Related Issue(s)
Related to https://github.com/gitpod-io/gitpod/issues/10651

## How to test
- Open a workspace in the preview environment
- Annotate the pod with gitpd.io/netConnLimit=true
- This should add additonal nftable rules to the network namespace of the pod (ssh into the node and then use `nsenter -t workspacekitPid -n` and `nft list ruleset`)
- Metrics can be obtained with `kubectl port-forward ds/ws-daemon 9500:9500,` then get the metrics with `curl -XGET localhost:9500/metrics`
- Run roblox scanner

## Release Notes

```release-note
Limit the rate at which network connections can be made by a workspace 
```
## Werft options:
